src/VisualStudio/Core/Def/Implementation/ProjectSystem/LinkedFileUtilities.cs

```diff
@@ -70,9 +70,7 @@ private IVsHierarchy GetSharedItemContextHierarchy(IVisualStudioHostDocument doc
                 return null;
             }
 
-            var sharedHierarchy = document.SharedHierarchy;
```

Comment:
It looks like `document.SharedHierarchy` should be removed, if you can't trust it's value.  Note that with asynchronous solution load there are cases in non-xproj shared projects where only one head may be loaded at the time the file is opened.